16-encoding-categorical-data.Rmd

```diff
@@ -1,8 +1,356 @@
 ```{r categorical-setup, include = FALSE}
 library(tidymodels)
+library(embed)
+library(textrecipes)
+library(kableExtra)
 tidymodels_prefer()
+source("ames_snippets.R")
+
+neighborhood_counts <- count(ames_train, Neighborhood)
 ```
 
 # (PART\*) Other Topics {-} 
 
 # Encoding categorical data  {#categorical}
+
+In Section \@ref(dummies) we introduced feature engineering approaches to encode or transform qualitative or nominal data into a representation better suited for most model algorithms. We discussed how to transform a categorical variable, such as the `Bldg_Type` in our Ames housing data (with levels `r knitr::combine_words(glue::backtick(levels(ames_train$Bldg_Type)))`), to a set of dummy or indicator variables.
+
+```{r engineering-all-dummies, echo = FALSE, results = 'asis'}
+show_rows <- 
+  ames_train %>% 
+  mutate(.row = row_number()) %>% 
+  group_by(Bldg_Type) %>% dplyr::select(Bldg_Type, .row) %>% 
+  slice(1) %>% 
+  pull(.row)
+recipe(~Bldg_Type, data = ames_train) %>% 
+  step_mutate(`Raw Data` = Bldg_Type) %>% 
+  step_dummy(Bldg_Type, naming = function(var, lvl, ordinal = FALSE, sep = "_") lvl) %>% 
+  prep() %>% 
+  bake(ames_train) %>% 
+  slice(show_rows) %>% 
+  arrange(`Raw Data`) %>% 
+  kable() %>% 
+  kable_styling(full_width = FALSE)
+```
+
+Many model implementations require such a transformation to a numeric representation for categorical data. 
```

Comment:
Should we mention that R prefers factors to represent these data and python typically uses integers?